birdbox/microsite/management/commands/bootstrap_footer.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import json
+import os
+from sys import stdout
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.db.transaction import atomic
+
+from wagtail.blocks import ListBlock
+from wagtail.blocks.list_block import ListValue
+from wagtail.core.rich_text import RichText
+
+from microsite.blocks import FooterSocialLinkBlock, LabelledLinkBlock
+from microsite.models import Footer
+
+
+def _print(*args):
+    stdout.write("\n".join(args) + "\n")
+
+
+class DryRunException(Exception):
+    pass
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--commit",
+            action="store_true",
+            help="Commit changes to the Footer to the database. Without this, does a dry run that confirms the footer config file is OK.",
+        )
+
+    def handle(self, *args, **options):
+        """
+        Delete the existing Footer setting and replace it with
+        a default set of content based on Mozilla.org
+        """
+        try:
+            self._rebuild_footer(inert=not options["commit"])
+        except DryRunException as e:
+            spacer = "*" * 64
+            _print(spacer, e.args[0], spacer)
+
+    def _pprint_data(self, sv):
+        collected = []
+        for block in sv:
+            for key, value in block.value.items():
+                if type(value) == ListValue:
+                    collected.append(f"{key}: \n")
+                    for val in value:
+                        collected.append(" ".join([f"{k}: {v} |" for k, v in val.items() if v]))
+                        collected.append("\n")
+                else:
+                    collected.append(f"\n{key}: {value}\n")
+
+        return "".join(collected)
+
+    @atomic
+    def _rebuild_footer(self, inert):
+        if inert:
+            _print("No --commit parameter passed. No changes will be stored")
+
+        result = Footer.objects.all().delete()
```

Comment:
Do you want the lack of `--commit` to also remove everything? That would go against the idea of a "dry run" and surprise me.

Reply:
Good q. Because `_rebuild_footer` is decorated with `@atomic` it should roll back the deletion as well as the addition. Seemed to work fine locally, too